cpp/src/io/parquet/page_enc.cu

```diff
@@ -1929,7 +1956,13 @@ dremel_data get_dremel_data(column_view h_col,
       curr_col = curr_col.child(0);
     }
     if (curr_col.type().id() == type_id::LIST) {
-      curr_col = curr_col.child(lists_column_view::child_column_index);
+      auto child = curr_col.child(lists_column_view::child_column_index);
+      if ((child.type().id() == type_id::INT8 || child.type().id() == type_id::UINT8) &&
+          output_as_byte_array) {
```

Comment:
No, we take the vector from the user and pass it straight into the gpu code. Should we make a verification pass first?

Reply:
AFAICT it would probably be slightly more optimal to check once/early on the host, but it's not a must ATM.

Reply:
I don't think we can get away with just verification on the host here. We need to find the last list column. It seems lots of places are hunting this and storing it. It is in the schema and the `column_device_view`. I would think a followup issue to investigate this would be in order.

Reply:
I left redundant comments about potentially redundant code. How the turn tables.
I guess my vision of clean logic is to check all columns marked as `output_as_byte_array` on the host/early on and proceed knowing that `output_as_byte_array` means that the column will definitely be output as such. If this does not sound like a good idea, feel free to resolve the convo :)

Reply:
This code isn't gate keeping the feature, but instead marching the children trying to find the leaf column. We change our opinion of what a leaf column is if we are outputting a byte_array. This code stops the leaf child marching to ensure the "leaf child" is actually the list, similar to a string column is considered a leaf column.